### PR TITLE
add defaulting to nats-cr in nats-manger

### PR DESCRIPTION
**Pull request by friedrichwilken**
**Description**

Changes proposed in this pull request:

adds defaulting to the nats cr:
```
spec:
  cluster:
    size: 3
  jetStream:
    fileStorage:
      size: 1Gi
      storageClassName: default
    memStorage:
      enabled: false
      size: 20Mi
  logging:
    debug: false
    trace: false
  resources:
    limits:
      cpu: 20m
      memory: 64Mi
    requests:
      cpu: 5m
      memory: 16Mi
```

**Related issue(s)**

### CODE REVIEW

File: api/v1alpha1/nats_types.go [hunk 1]
```diff
@@ -99,47 +126,56 @@ type NATSSpec struct {
...
+	// +kubebuilder:default:={storageClassName:"default",size:"1Gi"}
 	FileStorage `json:"fileStorage,omitempty"`
 }
 
 // MemStorage defines configurations to memory storage in NATS JetStream.
 type MemStorage struct {
-	// Enable allows the enablement of memory storage.
-	Enable bool `json:"enable"`
+	// Enabled allows the enablement of memory storage.
+	// +kubebuilder:default:=true
```
Comment (mfaizanse) [inline api/v1alpha1/nats_types.go:151]:
```suggestion
	// +kubebuilder:default:=false
```

Comment (mfaizanse) [reply api/v1alpha1/nats_types.go:151]: Please do `make manifest` if you accept this!

File: internal/controller/nats/integrationtests/validation/integration_test.go [hunk 2]
```diff
@@ -93,3 +99,70 @@ func Test_Validate_CreateNatsCR(t *testing.T) {
...
+				natsmatchers.HaveSpecJetsStreamMemStorage(v1alpha1.MemStorage{
+					Enabled: false,
+					Size:    resource.MustParse("20Mi"),
+				}),
+				natsmatchers.HaveSpecJetStreamFileStorage(v1alpha1.FileStorage{
+					StorageClassName: "default",
+					Size:             resource.MustParse("1Gi"),
+				}),
+			),
+		},
```
Comment (mfaizanse) [inline internal/controller/nats/integrationtests/validation/integration_test.go:147]: As per your comment:
```
Default values are defined at multiple levels to ensure that values are set independently of what level of NATS is
defined or gets deleted. For example, spec.cluster.size will get set to a default value whether spec.cluster.size,
spec.cluster or spec gets deleted.
```

Can we add more test cases to cover the multiple levels?

File: api/v1alpha1/nats_types.go [hunk 3]
```diff
@@ -38,28 +41,45 @@ const (
...
+to this topic at https://kubernetes.io/docs/reference/using-api/cel/.
+
+Testing of validation and defaulting is done via envtest in
+nats-manager/internal/controller/nats/integrationtests/validation/integration_test.go.
+
+For testing of defaulting, it is recommended to send an Unstructured object
+(https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured#Unstructured) to the API-Server. If
+non-nil-able properties like ResourceRequirements (https://pkg.go.dev/k8s.io/api/core/v1#ResourceRequirements) are
+undefined they will be interpreted as "" and result in 0 instead of being replaced by the default value.
+*/
```
Comment (mfaizanse) [inline api/v1alpha1/nats_types.go:71]: Thumbs up for adding the comments!!!

File: internal/controller/nats/integrationtests/validation/integration_test.go (hunk 2 above)
Comment (friedrichwilken) [reply internal/controller/nats/integrationtests/validation/integration_test.go:147]: valid! will do!

**Review by mfaizanse — APPROVED** [review]

**Pull request merged**